src/rpc/core_rpc_server.cpp

```diff
@@ -1437,7 +1437,7 @@ namespace cryptonote
         return false;
       }
       b.nonce = req.starting_nonce;
-      miner::find_nonce_for_given_block(b, template_res.difficulty, template_res.height);
+      miner::find_nonce_for_given_block(reinterpret_cast<i_miner_handler*>(&m_core), b, template_res.difficulty, template_res.height);
```

Comment:
This cast should not be needed, is it actually needed ?

Reply:
Ah you're right. I never even tried it without the cast.